connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -481,8 +484,41 @@ public boolean confirmTeleport(Vector3d position) {
             int teleportId = teleportCache.getTeleportConfirmId();
             teleportCache = null;
             ClientTeleportConfirmPacket teleportConfirmPacket = new ClientTeleportConfirmPacket(teleportId);
-            getDownstream().getSession().send(teleportConfirmPacket);
+            sendDownstreamPacket(teleportConfirmPacket);
         }
         return true;
     }
+
+    /**
+     * Queue a packet to be sent to player.
+     *
+     * @param packet the bedrock packet from the NukkitX protocol lib
+     */
+    public void sendUpstreamPacket(BedrockPacket packet) {
+        if (upstream != null && !upstream.isClosed()) {
```

Comment:
Could we add debug logs here if upstream is closed or null? Same with the other methods.